src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -4,45 +4,72 @@
     {
         private static readonly string s_defaultCachePath = Path.Combine(
             Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),
             ".docfx",
             "cache",
             "user-profile.json");
 
-        private readonly ConcurrentDictionary<string, GitRepoInfo> _folderRepoInfocache
-            = new ConcurrentDictionary<string, GitRepoInfo>();
+        private readonly ConcurrentDictionary<string, GitRepoInfo> _folderRepoInfocache = new ConcurrentDictionary<string, GitRepoInfo>();
 
-        private readonly Lazy<Dictionary<string, List<GitCommit>>> _fileCommitsCache;
+        private readonly UserProfileCache _userProfileCache;
 
-        private readonly GitUserProfileCache _githubUserProfileCache;
+        private readonly IReadOnlyDictionary<string, List<GitCommit>> _commitsByFile;
 
-        private readonly ReadOnlyDictionary<string, DateTime> _commitTimeBySha;
+        private readonly IReadOnlyDictionary<string, DateTime> _updateTimeByCommit;
 
-        public GitRepoInfoProvider(Docset docset)
+        public ContributionInfo(
+            IReadOnlyDictionary<string, List<GitCommit>> commitsByFile,
+            IReadOnlyDictionary<string, DateTime> updateTimeByCommit,
+            UserProfileCache userProfileCache)
         {
-            Debug.Assert(docset != null);
+            _commitsByFile = commitsByFile;
+            _updateTimeByCommit = updateTimeByCommit;
+            _userProfileCache = userProfileCache;
+        }
+
+        public static ContributionInfo Load(Docset docset)
+        {
+            var commitsByFile = LoadCommits(docset);
+
+            var updateTimeByCommit = string.IsNullOrEmpty(docset.Config.Contribution.GitCommitsTimePath)
+                ? new Dictionary<string, DateTime>()
+                : GitCommitsTime.Create(Path.Combine(docset.DocsetPath, docset.Config.Contribution.GitCommitsTimePath)).ToDictionary();
 
-            _fileCommitsCache = new Lazy<Dictionary<string, List<GitCommit>>>(() => LoadCommits(docset));
             var userProfileCachePath = string.IsNullOrEmpty(docset.Config.Contribution.UserProfileCachePath)
                 ? s_defaultCachePath
                 : Path.Combine(docset.DocsetPath, docset.Config.Contribution.UserProfileCachePath);
-            _githubUserProfileCache = GitUserProfileCache.Create(userProfileCachePath);
 
-            _commitTimeBySha = new ReadOnlyDictionary<string, DateTime>(
-                string.IsNullOrEmpty(docset.Config.Contribution.GitCommitsTimePath)
-                ? new Dictionary<string, DateTime>()
-                : GitCommitsTime.Create(Path.Combine(docset.DocsetPath, docset.Config.Contribution.GitCommitsTimePath)).ToDictionary());
+            return new ContributionInfo(commitsByFile, updateTimeByCommit, UserProfileCache.Create(userProfileCachePath));
+        }
+
+        private static IReadOnlyDictionary<string, List<GitCommit>> LoadCommits(Docset docset)
```

Comment:
>private [](start = 8, length = 7)

move private methods below?

Reply:
this is a static method, so I first group them by static/instance then by public/private

---
In reply to: [199416369](https://github.com/dotnet/docfx/pull/2981#discussion_r199416369) [](ancestors = 199416369)